dffml/operation/db.py

```diff
@@ -0,0 +1,189 @@
+
+    try:
+        await query_fn(**kwargs)
+        return {"lookups":{}}
+
+    except TypeError as e:
+        if 'async_gen' in repr(e):
+            result = query_fn(**kwargs)
+            return {"lookups":[res async for res in result] }
+        else :
+            raise e
+
+
+@op(
+    inputs={
+        "table_name":QUERY_TABLE,
+        "cols":QUERY_COLS
+    },
+    outputs={
+        "lookups":QUERY_LOOKUPS
+    },
+    config_cls=SqliteQueryConfig,
+    imp_enter={"database": (lambda self: self.config.database)},
+    ctx_enter={"dbctx": (lambda self: self.parent.database())},
+)
+async def sqlite_query_create_table(self,*,
+                table_name : str,
+                cols : List[str] = [],
+                ) -> Dict[str, Any]:
+                await self.dbctx.create_table(table_name=table_name,
+                                        cols=cols)
+                return {"lookups":{}}
+
+@op(
+    inputs={
+        "table_name":QUERY_TABLE,
+        "data":QUERY_DATA
+    },
+    outputs={
+        "lookups":QUERY_LOOKUPS
+    },
+    config_cls=SqliteQueryConfig,
+    imp_enter={"database": (lambda self: self.config.database)},
+    ctx_enter={"dbctx": (lambda self: self.parent.database())},
+)
+async def sqlite_query_insert(self,*,
+                table_name : str,
+                data : Dict[str,Any],
+                ) -> Dict[str, Any]:
+                await self.dbctx.insert(table_name=table_name,
+                                        data=data)
+                return {"lookups":{}}
+
+@op(
+    inputs={
+        "table_name":QUERY_TABLE,
+        "data":QUERY_DATA,
+        "conditions":QUERY_CONDITIONS
+    },
+    outputs={
+        "lookups":QUERY_LOOKUPS
+    },
+    config_cls=SqliteQueryConfig,
+    imp_enter={"database": (lambda self: self.config.database)},
+    ctx_enter={"dbctx": (lambda self: self.parent.database())},
+)
+async def sqlite_query_update(self,*,
+                table_name : str,
+                data : Dict[str,Any],
+                conditions : Conditions = []
+                ) -> Dict[str, Any]:
+                await self.dbctx.update(table_name=table_name,
+                                        data=data,
+                                        conditions=conditions)
+                return {"lookups":{}}
+
+@op(
+    inputs={
+        "table_name":QUERY_TABLE,
+        "conditions":QUERY_CONDITIONS
+    },
+    outputs={
+        "lookups":QUERY_LOOKUPS
+    },
+    config_cls=SqliteQueryConfig,
+    imp_enter={"database": (lambda self: self.config.database)},
+    ctx_enter={"dbctx": (lambda self: self.parent.database())},
+)
+async def sqlite_query_remove(self,*,
+                table_name : str,
+                conditions : Conditions = []
+                ) -> Dict[str, Any]:
+                await self.dbctx.remove(table_name=table_name,
+                                        conditions=conditions)
+                return {"lookups":{}}
```

Comment:
I can't remember, was there a reason we wanted to return lookups from the non-lookup operations still? Or was this just an artifact of the literal translation of spiting the operation into multiple operations?

Reply:
My bad,copy paste errors.removed it.